### DISCUSSION TITLE
Checking for private key

### DISCUSSION THREAD

**Discussion by afshinpir** [openssl/openssl#24268] [Q&A]
Hi all

How I can check an `EVP_PKEY` contains a private key or not? I cannot find anything corresponding to provider key management's `_has()` method from public API.

**Comment by kulkarniamit** [comment]
You can use [EVP_PKEY_private_check()](https://www.openssl.org/docs/man3.0/man3/EVP_PKEY_private_check.html) to validate the private component of the key given by ctx.
```c
    ...
    EVP_PKEY *pkey = ...
    ...
    EVP_PKEY_CTX *ctx = EVP_PKEY_CTX_new(pkey, NULL);
    if (!ctx) {
        fprintf(stderr, "Failed to get a ctx\n");
        return 1;
    }
    fprintf(stdout, "pkey is a %s key\n",
        EVP_PKEY_private_check(ctx) ? "private":"public");
   ...
```
